smart-contracts/contracts/mixins/MixinPurchase.sol

```diff
@@ -86,14 +93,24 @@ contract MixinPurchase is
       // This is an existing owner trying to extend their key
       newTimeStamp = toKey.expirationTimestamp + expirationDuration;
       toKey.expirationTimestamp = newTimeStamp;
+
+      // set key manager
+      if(_keyManager != address(0) && keyManagerOf[idTo] != _keyManager) {
+          _setKeyManagerOf(idTo, _keyManager);
+      }
+
       emit RenewKeyPurchase(_recipient, newTimeStamp);
     } else {
       // This is an existing owner trying to renew their expired key
       newTimeStamp = block.timestamp + expirationDuration;
       toKey.expirationTimestamp = newTimeStamp;
 
-      // reset the key Manager to 0x00
-      _setKeyManagerOf(idTo, address(0));
+      if(_keyManager != address(0) && keyManagerOf[idTo] != _keyManager) {
+          _setKeyManagerOf(idTo, _keyManager);
+      } else {
```

Comment:
I don't think we need this anymore. Basically whoever will "pay" for that key (call the `purchase` function) will get to decide who the manager is and I think that's fine.

Reply:
you mean not reseting the key manager to address zero?

also one possible case is someone willing to extend their key and keep the existing key manager. We could check id a key manager already exists if address zero is passed, so we just keep the existing key manager, instead of erasing it ?

Reply:
I think the best way is to just set it to whatever the caller of the `purchase` transaction decided. This will bring clarity and make things easier to manager.